hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/compact/FlinkScheduleCompactionActionExecutor.java

```diff
@@ -149,4 +155,41 @@ public Long parsedToSeconds(String time) {
     }
     return timestamp;
   }
+
+  @Override
+  public Option<HoodieCompactionPlan> execute() {
+    if (!config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()
+            && !config.getFailedWritesCleanPolicy().isLazy()) {
+      // if there are inflight writes, their instantTime must not be less than that of compaction instant time
+      table.getActiveTimeline().getCommitsTimeline().filterPendingExcludingCompaction().firstInstant()
```

Comment:
should we move these validation checks to a commonplace? Looks like this was being used elsewhere too.

Reply:
There is a bug in BaseScheduleCompactionActionExecutor$execute as the PR https://github.com/apache/hudi/pull/3025 want to fix, this bug will let independence hudi compaction can not run. But this PR need to discuss, so we choose to modify the subclass FlinkScheduleCompactionActionExecutor to finish compaction.